encoding/protoavro/marshal.go

```diff
@@ -28,30 +60,29 @@ func NewMarshaler(descriptor protoreflect.MessageDescriptor, writer io.Writer) (
 	if err != nil {
 		return nil, fmt.Errorf("new ocf writer: %w", err)
 	}
-	return &Marshaler{w: w, desc: descriptor}, nil
+	return &Marshaler{w: w, desc: descriptor, options: options}, nil
 }
 
 // Marshaler encodes and writes Avro binary encoded messages.
 type Marshaler struct {
 	desc protoreflect.MessageDescriptor
 	w    *goavro.OCFWriter
+	options *MarshalOptions
 }
 
 // Marshal encodes and writes messages to the writer.
-func (m *Marshaler) Marshal(messages ...proto.Message) error {
-	data := make([]interface{}, 0, len(messages))
-	for _, message := range messages {
-		a := message.ProtoReflect().Descriptor().FullName()
-		b := m.desc.FullName()
-		if a != b {
-			return fmt.Errorf("expected message '%s' but got '%s'", a, b)
-		}
-		m, err := encodeJSON(message)
-		if err != nil {
-			return fmt.Errorf("encode json: %w", err)
-		}
-		data = append(data, m)
+func (m *Marshaler) Marshal(message proto.Message) error {
```

Comment:
This method should take the `MarshalOptions` (not use the Marshal "global" options).